inc/azure_c_logging/xlogging.h

```diff
@@ -87,14 +87,22 @@ typedef void(*LOGGER_LOG_GETLASTERROR)(const char* file, const char* func, int l
 
 #define LOG(log_category, log_options, format, ...) MU_C2(LOG_,log_category)(log_category, log_options, format, ##__VA_ARGS__)
 
+#ifdef _MSC_VER
+static __declspec(thread) char stackAsString[STACK_MAX_CHARACTERS];
+static __declspec(thread) char formatWithStack[FORMAT_MAX_CHARACTERS];
+#else
+/*for C11 compilers*/
+static _Thread_local char stackAsString[STACK_MAX_CHARACTERS];
+static _Thread_local char formatWithStack[FORMAT_MAX_CHARACTERS];
```

Comment:
Consider a 
#ifdef _MSC_VER
#define XLOGGING_THREAD_LOCAL __declspec(thread)
#elif /* check for C11 compiler */
#define XLOGGING_THREAD_LOCAL _Thread_local
#else
#error
#endif

And then having the variables only once:

static XLOGGING_THREAD_LOCAL char stackAsString[STACK_MAX_CHARACTERS];
static XLOGGING_THREAD_LOCAL char formatWithStack[FORMAT_MAX_CHARACTERS];

There are actually 2 suggestions in here:
- Have a #define with the "thread local" modifier
- Error out at compile time of the compiler is not C11

Reply:
I have the define. And using it.

I am not sure why a separate check would be needed for the other compilers: they would either understand _Thread_local C keyword or barf at compilation time anyway. Right?

---
In reply to: [504393960](https://github.com/Azure/azure-c-logging/pull/18#discussion_r504393960) [](ancestors = 504393960)